### PR TITLE
OCPBUGS-24195: the apiservice controller waits until bootstrap complete

### PR DESCRIPTION
**Pull request by p0lyn0mial**
"fixes" the issue described in https://github.com/openshift/oauth-apiserver/pull/105

The number of `OpenShiftAPICheckFailed` generated by the operator with the fix for `e2e-operator` run:
Screenshot 2024-04-15 at 11 34 13

Now, without fix, taken from [a previous e2e-operator](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-authentication-operator/664/pull-ci-openshift-cluster-authentication-operator-master-e2e-operator/1778329477852434432) run:
Screenshot 2024-04-15 at 11 35 19

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -638,12 +638,13 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
...
 	).WithAPIServiceController(
 		"openshift-apiserver",
+		"openshift-oauth-apiserver",
 		func() ([]*apiregistrationv1.APIService, []*apiregistrationv1.APIService, error) {
 			return apiServices(), nil, nil
 		},
 		apiregistrationInformers,
 		apiregistrationv1Client.ApiregistrationV1(),
-		operatorCtx.kubeInformersForNamespaces.InformersFor("openshift-oauth-apiserver"),
+		operatorCtx.kubeInformersForNamespaces,
```
Comment (stlaz) [inline pkg/operator/starter.go:647]: This seems wrong. Why do you change it?

Comment (p0lyn0mial) [reply pkg/operator/starter.go:647]: I changed the syntax of this method in https://github.com/openshift/library-go/pull/1694

**Comment by p0lyn0mial** [comment]
> pinning and bump in a single commit, please

done

**Comment by stlaz** [comment]
> done
I still see it as 2 commits

**Comment by p0lyn0mial** [comment]
> I still see it as 2 commits

sorry, fixed now.

**Comment by p0lyn0mial** [comment]
/retest-required

`e2e-operator` fails on flaky `TestKeycloakAsOIDCPasswordGrantCheckAndGroupSync` - https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-openshift-cluster-authentication-operator-master-e2e-operator

**Pull request merged**